src/main/java/org/cryptomator/cryptofs/fh/OpenCryptoFile.java

```diff
@@ -105,6 +96,20 @@ public synchronized FileChannel newFileChannel(EffectiveOpenOptions options, Fil
 		return cleartextFileChannel;
 	}
 
+	//visible for testing
+	void initFileHeader(EffectiveOpenOptions options, FileChannel ciphertextFileChannel) throws IOException {
+		if (headerHolder.get() == null) {
+			//first file channel to file, no header present
+			if (options.createNew() || (options.create() && ciphertextFileChannel.size() == 0)) {
+				//file did not exist, create new header
+				headerHolder.createNew();

```

Comment:
>In that case I believe we can eagerly write the header, can't we? 

I also thought about it, but since i could not find the reason for the lazy header writing, i decided to leave it as it is.

Reply:
There exists an issue regarding writing the header too often (https://github.com/cryptomator/cryptofs/issues/58) or not sharing the file header (https://github.com/cryptomator/cryptofs/issues/48).

But i cannot not find any info, why we started writing the header lazily

Reply:
Let's better keep it that way until we can be sure about the effects of eagerly writing the header.